File: extract_bottles.js
import sharp from 'sharp';

async function extractBottlesAndBackgrounds() {
    const perfumes = [
        { name: 'GIT', file: 'GIT.png' },
        { name: 'Ganymede', file: 'Ganymede.png' },
        { name: 'Guidance', file: 'Guidance.png' }
    ];

    for (const perfume of perfumes) {
        console.log(`Processing ${perfume.name}...`);

        const inputPath = `public/images/${perfume.file}`;

        // Get image metadata
        const metadata = await sharp(inputPath).metadata();
        console.log(`  Original size: ${metadata.width}x${metadata.height}`);

        // Extract just the bottle - keep more of the image this time
        // The bottles are in the center, we'll crop less aggressively
        const bottleWidth = Math.floor(metadata.width * 0.7);  // Increased from 0.5
        const bottleHeight = Math.floor(metadata.height * 0.9); // Increased from 0.8
        const bottleLeft = Math.floor((metadata.width - bottleWidth) / 2);
        const bottleTop = Math.floor(metadata.height * 0.05); // Start higher

        await sharp(inputPath)
            .extract({
                left: bottleLeft,
                top: bottleTop,
                width: bottleWidth,
                height: bottleHeight
            })
            .resize(600, 800, {  // Larger output size
                fit: 'contain',
                background: { r: 0, g: 0, b: 0, alpha: 0 }
            })
            .toFile(`public/images/${perfume.name}-bottle.png`);

        console.log(`  ✓ Created bottle: ${perfume.name}-bottle.png`);

        // Create background from the full image (we'll use it as background)
        await sharp(inputPath)
            .blur(15) // Blur for background effect
            .modulate({
                brightness: 0.8,  // Darken slightly
                saturation: 1.2   // Increase saturation
            })
            .toFile(`public/images/${perfume.name}-bg.png`);

        console.log(`  ✓ Created background: ${perfume.name}-bg.png`);
    }

    console.log('\n✅ All bottles and backgrounds extracted!');
}

extractBottlesAndBackgrounds().catch(console.error);

File: extract_bottles.py
from PIL import Image
import os

# Process each perfume bottle image
images = ['GIT.png', 'Ganymede.png', 'Guidance.png']
input_dir = '/Users/diwineoperator/Documents/perfume/public/images'
output_dir = '/Users/diwineoperator/Documents/perfume/public/images'

for img_name in images:
    input_path = os.path.join(input_dir, img_name)
    
    # Load the image
    img = Image.open(input_path)
    
    # Convert to RGBA if not already
    if img.mode != 'RGBA':
        img = img.convert('RGBA')
    
    # Get the bottle region (center portion)
    width, height = img.size
    
    # Extract the bottle with transparency (center region)
    # The bottles appear to be in the center of the image
    bottle_left = int(width * 0.25)
    bottle_right = int(width * 0.75)
    bottle_top = int(height * 0.1)
    bottle_bottom = int(height * 0.9)
    
    # Crop to bottle area
    bottle = img.crop((bottle_left, bottle_top, bottle_right, bottle_bottom))
    
    # Save the extracted bottle
    bottle_output = os.path.join(output_dir, img_name.replace('.png', '-bottle.png'))
    bottle.save(bottle_output, 'PNG')
    
    # Create background from the abstract art (the colorful background area)
    # Sample the background area and create a larger pattern
    bg_sample = img.copy()
    
    # Save the background
    bg_output = os.path.join(output_dir, img_name.replace('.png', '-bg.png'))
    bg_sample.save(bg_output, 'PNG')
    
    print(f"Processed {img_name}")
    print(f"  - Bottle: {bottle_output}")
    print(f"  - Background: {bg_output}")

print("\nAll images processed!")
